banzai_nres/extract.py

```diff
@@ -25,8 +29,10 @@ def do_stage(self, image: EchelleSpectralCCDData):
             x_extent = slice(np.min(this_trace[1]), np.max(this_trace[1]) + 1)
             flux[i, x_extent] = self.extract_order(image.data[this_trace], image.weights[this_trace])
             variance[i, x_extent] = self.extract_order(image.uncertainty[this_trace] ** 2, image.weights[this_trace] ** 2)
+            # get the average wavelength: Sum wavelengths weighted by 1 over the vertical width of the trace (e.g. 1/10)
+            wavelength[i, x_extent] = self.extract_order(image.wavelengths[this_trace], weights=1/image.wavelengths[this_trace].shape[0])
```

Comment:
This feels like a weird place for this logic. But I'm not sure where else to put it.

Reply:
I agree.